File: push-openapi-python-demo/Custom_Signature/request.py
from sign import *
def request(host,params,secret):
    param_string = ''
    for k in sorted(params.keys()):
        param_string += urllib.quote(k) + "=" + urllib.quote(params[k]) + "&"
    request = host + "/?" + param_string + "Signature=" + sign("GET",params,secret+'&')
    return request

File: OpenApi1.0/push-openapi-php-demo/Custom_Signature/request.php
<?php   
	include "sign.php";
	function request($host,$params,$secret)
	{
		$param_string = "";
		foreach($params as $key => $value)
		{
			$param_string .= urlencode($key) . "=" . urlencode($value) . "&"; 
		}
		return  $host . "/?" . $param_string . "Signature=" . sign("GET",$params,$secret . "&");
	}
?>
